.github/workflows/node-build.yml

```diff
@@ -0,0 +1,309 @@
+name: Node Build
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - '**'
+  push:
+    branches:
+      - main
+      - release/v*
+
+jobs:
+  prerequisite:
+    runs-on: ubuntu-latest
+    timeout-minutes: 1
+    steps:
+    - uses: actions/checkout@v4
+    - uses: ./.github/workflows/rafiki/env-setup
+    - run: pnpm checks
+
+  backend:
+    runs-on: ubuntu-latest
+    needs: prerequisite
+    timeout-minutes: 25
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/workflows/rafiki/env-setup
+      - run: pnpm --filter backend build:deps
+      - run: NODE_OPTIONS=--max-old-space-size=4096 pnpm --filter backend test:ci
+
+  frontend:
+    runs-on: ubuntu-latest
+    needs: prerequisite
+    timeout-minutes: 5
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/workflows/rafiki/env-setup
+      - run: pnpm --filter frontend typecheck
+      - run: pnpm --filter frontend build
+
+  auth:
+    runs-on: ubuntu-latest
+    needs: prerequisite
+    timeout-minutes: 5
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/workflows/rafiki/env-setup
+      - run: pnpm --filter auth build:deps
+      - run: pnpm --filter auth test
+      - name: AsyncAPI extension
+        run: |
+          echo "{\"extends\":[\"spectral:oas\",\"spectral:asyncapi\"]}" >> .spectral.json
+      - name: Validate Open API specs
+        run: |
+          npx @stoplight/spectral-cli lint ./packages/auth/openapi/*.yaml
```

Comment:
I think this section also needs to be added to backend. I'm aware of the fact that it may not have been there before, but I'm spotting it now 😉